modules/src/events.rs

```diff
@@ -328,6 +336,73 @@ impl TryFrom<IbcEvent> for AbciEvent {
     }
 }
 
+impl TryFrom<&AbciEvent> for IbcEvent {
```

Comment:
Same here, I think it would be better to have this code in the relayer crate. If the orphan rule precludes us from moving it there, we could have an `impl TryFrom<AbciEvent> for IbcEventWithHeight` along with a `impl From<IbcEventWithHeight> for IbcEvent` in the relayer crate. WDYT?

Reply:
I indeed put it there because of the orphan rule. I don't like `impl TryFrom<AbciEvent> for IbcEventWithHeight` because there is never any `height` information in the `AbciEvent`, so you can never build an `IbcEventWithHeight` from it. While I understand that what you're suggesting is that we should always do `AbciEvent` -> `IbcEventWithHeight` -> `IbcEvent`, I find it hacky for the reason explained above.

I thought about putting it behind a `"relayer"` feature flag. WDYT?

Reply:
I like the relayer feature flag idea. :+1: But I am concerned about the light client refactoring and `UpdateClientEvent`'s header field; keeping it in the relayer allows us to use `AnyHeader` there (in `extract_header_from_tx`). Would it be possible to use a `impl TryFrom<(AbciEvent, Height)> for IbcEventWithHeight` instead and still keep this code in the relayer crate?

Reply:
I agree, `ibc_event_try_from_abci_event()` seems like the best solution. :+1:

Reply:
[785cbad](https://github.com/informalsystems/ibc-rs/pull/2542/commits/785cbada83692347edd63dc1ee9dfa54a50505b6) & [1bdf27f](https://github.com/informalsystems/ibc-rs/pull/2542/commits/1bdf27f4710f8e65c0bb37deaff679e4f519715e)